c_glib/arrow-glib/table.cpp

```diff
@@ -342,7 +342,28 @@ garrow_table_equal(GArrowTable *table, GArrowTable *other_table)
 {
   const auto arrow_table = garrow_table_get_raw(table);
   const auto arrow_other_table = garrow_table_get_raw(other_table);
-  return arrow_table->Equals(*arrow_other_table);
+  return arrow_table->Equals(*arrow_other_table, false);
+}
+
+/**
+ * garrow_table_equal_metadata:
+ * @table: A #GArrowTable.
+ * @other_table: A #GArrowTable to be compared.
+ * @metadata: Whether to compare metadata.
```

Comment:
@shiro615 Sorry... I reconsider this too. `@check_metadata` will be better name. What do you think about this?

Reply:
Thank you for reconsidering this. I think so too.  I'll use `check_metadata`.

Reply:
I've changed this parameter name.